gpkit/geometric_program.py

```diff
@@ -148,9 +152,41 @@ def solve(self, solver=None, verbosity=1, *args, **kwargs):
             print("Solving took %.3g seconds." % (soltime,))
             tic = time()
 
+        if solver_out.get("status", None) not in ["optimal", "OPTIMAL"]:
+            raise RuntimeWarning(
+                "final status of solver '%s' was '%s', not 'optimal'."
+                "\n\nThe infeasible solve's result is stored in"
+                " model.program.solver_out. A result dict can be generated "
+                " via program._compile_result(program.solver_out)."
+                " If the problem was Primal Infeasible, you can generate a"
+                " feasibility-finding relaxation with model.feasibility()." %
+                (solver, solver_out.get("status", None)))
+
+        result = self._compile_result(solver_out)
+
+        self.result = result  # NOTE: SIDE EFFECTS
+
+        if verbosity > 1:
+            print ("result packing took %.2g%% of solve time" %
+                   ((time() - tic) / soltime * 100))
+            tic = time()
+
+        np.ravel(solver_out['primal'])
+        self.check_solution(result["cost"],
+                            np.ravel(solver_out['primal']),
+                            nu=result["sensitivities"]["monomials"],
+                            la=result["sensitivities"]["posynomials"])
+
+        if verbosity > 1:
+            print ("solution checking took %.2g%% of solve time" %
+                   ((time() - tic) / soltime * 100))
+        return result
+
+    def _compile_result(self, solver_out):
```

Comment:
right -- by making this a function we can avoid calling it when a solution is not optimal -- and we leave it up the the user whether they want to try passing `solver_out` into `_compile_result` -- with exp overflows being one possible result

My feeling is that for a user that is not debugging, getting an exp overflow feels icky.